instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/TracerTest.java

```diff
@@ -0,0 +1,329 @@
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import io.opentelemetry.semconv.ExceptionAttributes;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@ExtendWith(AgentInstrumentationExtension.class)
+public class TracerTest {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @Test
+  @DisplayName("capture span, kind, attributes, and status")
+  public void captureSpanKindAttributesAndStatus() {
+    // When
+    Tracer tracer = GlobalOpenTelemetry.getTracer("test");
+    Span testSpan = tracer.spanBuilder("test").setSpanKind(PRODUCER).startSpan();
+    testSpan.setAttribute("string", "1");
+    testSpan.setAttribute("long", 2L);
+    testSpan.setAttribute("double", 3.0);
+    testSpan.setAttribute("boolean", true);
+    testSpan.setStatus(ERROR);
+    testSpan.end();
+
+    // Then
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("test")
+                        .hasKind(PRODUCER)
+                        .hasNoParent()
+                        .hasStatus(StatusData.error())
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(stringKey("string"), "1"),
+                            equalTo(longKey("long"), 2L),
+                            equalTo(doubleKey("double"), 3.0),
+                            equalTo(booleanKey("boolean"), true))));
+  }
+
+  @Test
+  @DisplayName("capture span with implicit parent using Tracer.withSpan()")
+  public void captureSpanWithImplicitParentUsingTracerWithSpan() {
+    // When
+    Tracer tracer = GlobalOpenTelemetry.getTracer("test");
+    Span parentSpan = tracer.spanBuilder("parent").startSpan();
+    Scope parentScope = Context.current().with(parentSpan).makeCurrent();
+
+    Span testSpan = tracer.spanBuilder("test").startSpan();
+    testSpan.end();
+
+    parentSpan.end();
+    parentScope.close();
+
+    // Then
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span0 -> span0.hasName("parent").hasNoParent().hasAttributesSatisfyingExactly(),
+                span1 ->
```

Comment:
each one of these could simply be named `span`, you don't need to add a numerical suffix